src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -0,0 +1,324 @@
+                       .Start(Assembly.GetCallingAssembly(), mdToken, opCode)
+                       .WithConcreteTypeName(DiagnosticSource)
+                       .WithParameters(diagnosticSource, actionDescriptor, httpContext, routeData)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                // profiled app will continue working as expected without this method
+                Log.ErrorException($"Error resolving {methodDef}", ex);
+            }
+
+            try
+            {
+                // call the original method, catching and rethrowing any unhandled exceptions
+                instrumentedMethod?.Invoke(diagnosticSource, actionDescriptor, httpContext, routeData);
+            }
+            catch (Exception ex)
+            {
+                // profiled app will continue working as expected without this method
+                Log.Error($"Exception when calling {methodDef}.", ex);
+            }
+            finally
+            {
+                aspNetCoreMvcActionScope?.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Wrapper method used to catch unhandled exceptions in the incoming request pipeline for Microsoft.AspNetCore.Mvc.Core
+        /// </summary>
+        /// <param name="context">The DiagnosticSource that this extension method was called on.</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        [InterceptMethod(
+            CallerAssembly = AspnetMvcCore,
+            TargetAssembly = AspnetMvcCore,
+            TargetType = ResourceInvoker,
+            TargetSignatureTypes = new[] { ClrNames.Void, ClrNames.Ignore },
+            TargetMinimumVersion = Major2,
+            TargetMaximumVersion = Major2)]
+        public static void Rethrow(object context, int opCode, int mdToken)
+        {
+            string methodDef = $"{ResourceInvoker}.{nameof(Rethrow)}({context?.GetType().FullName} context)";
+            var shouldTrace = Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName);
+
+            Action<object> instrumentedMethod = null;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Action<object>>
+                       .Start(Assembly.GetCallingAssembly(), mdToken, opCode)
+                       .WithConcreteTypeName(ResourceInvoker)
+                       .WithParameters(context)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error retrieving {methodDef}", ex);
+                throw;
+            }
+
+            AspNetAmbientContext ambientContext = null;
+            var exceptionToGrab = context.GetProperty<Exception>("Exception");
```

Comment:
In one of the previous bugs, `context` was null. Can that happen here too?

Reply:
Good catch, I think it would make sense to add a null check within this extension method. 
What are your thoughts on that?

Reply:
Agreed, checking within the extension method seems like a good approach.